src/libsystemd/sd-netlink/netlink-types-rtnl.c

```diff
@@ -940,13 +942,13 @@ static const NLType rtnl_route_metrics_types[] = {
 DEFINE_TYPE_SYSTEM(rtnl_route_metrics);
 
 static const NLType rtnl_route_types[] = {
-        [RTA_DST]               = { .type = NETLINK_TYPE_IN_ADDR }, /* 6? */
-        [RTA_SRC]               = { .type = NETLINK_TYPE_IN_ADDR }, /* 6? */
+        [RTA_DST]               = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) }, /* 6? */
+        [RTA_SRC]               = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) }, /* 6? */
         [RTA_IIF]               = { .type = NETLINK_TYPE_U32 },
         [RTA_OIF]               = { .type = NETLINK_TYPE_U32 },
-        [RTA_GATEWAY]           = { .type = NETLINK_TYPE_IN_ADDR },
+        [RTA_GATEWAY]           = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) },
         [RTA_PRIORITY]          = { .type = NETLINK_TYPE_U32 },
-        [RTA_PREFSRC]           = { .type = NETLINK_TYPE_IN_ADDR }, /* 6? */
+        [RTA_PREFSRC]           = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) }, /* 6? */
```

Comment:
All these take both IPv4 and IPv6. Please drop the change.

Reply:
Dropped